qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,1037 @@
+            schedule: The schedule or its name for which to get the parameter value.
+            valid_only: Use only parameters marked as valid.
+            group: The calibration group from which to draw the parameters.
+                If not specified this defaults to the 'default' group.
+            cutoff_date: Retrieve the most recent parameter up until the cutoff date. Parameters
+                generated after the cutoff date will be ignored. If the cutoff_date is None then
+                all parameters are considered. This allows users to discard more recent values that
+                may be erroneous.
+
+        Returns:
+            value: The value of the parameter.
+
+        Raises:
+            CalibrationError:
+                - If there is no parameter value for the given parameter name and pulse channel.
+        """
+        qubits = self._to_tuple(qubits)
+
+        # 1) Identify the parameter object.
+        param_name = param.name if isinstance(param, Parameter) else param
+        sched_name = schedule.name if isinstance(schedule, ScheduleBlock) else schedule
+
+        param = self.calibration_parameter(param_name, qubits, sched_name)
+
+        # 2) Get a list of candidate keys restricted to the qubits of interest.
+        candidate_keys = []
+        for key in self._parameter_map_r[param]:
+            candidate_keys.append(ParameterKey(key.parameter, qubits, key.schedule))
+
+        # 3) Loop though the candidate keys to candidate values
+        candidates = []
+        for key in candidate_keys:
+            if key in self._params:
+                candidates += self._params[key]
+
+        # If no candidate parameter values were found look for default parameters
+        # i.e. parameters that do not specify a qubit.
+        if len(candidates) == 0:
+            for key in candidate_keys:
+                if ParameterKey(key.parameter, (), key.schedule) in self._params:
+                    candidates += self._params[ParameterKey(key.parameter, (), key.schedule)]
+
+        # 4) Filter candidate parameter values.
+        if valid_only:
+            candidates = [val for val in candidates if val.valid]
+
+        candidates = [val for val in candidates if val.group == group]
+
+        if cutoff_date:
+            candidates = [val for val in candidates if val.date_time <= cutoff_date]
+
+        if len(candidates) == 0:
+            msg = f"No candidate parameter values for {param_name} in calibration group {group} "
+
+            if qubits:
+                msg += f"on qubits {qubits} "
+
+            if sched_name:
+                msg += f"in schedule {sched_name} "
+
+            if cutoff_date:
+                msg += f"with cutoff date: {cutoff_date}"
+
+            raise CalibrationError(msg)
+
+        # 5) Return the most recent parameter.
+        return max(candidates, key=lambda x: x.date_time).value
+
+    def get_schedule(
```

Comment:
To me still `get_schedule` and `get_template` seem to be confusing. Is it possible to add a context of _calibrated_ to the method name? i.e. `get_schedule` is mainly used to get a schedule that is partly or fully calibrated, while `get_template` is used to create new schedule template to be calibrated. Or we can integrate `get_template` as an option of `get_schedule`, i.e. `return_template: bool`.

The main point of this comment is::
```python
with build() as ecr:
    play(GaussianSquare(***), u0)
    call(my_cal.get_schedule('xp', (0,)))
    play(GaussianSquare(***), u0)
```
doesn't look strange to me. Perhaps we don't need `get_template` method because the template of called subroutine should exist in the manager. This means `add_schedule` method can handle call instruction as follows; If it detects the call, this can internally get the parameter unassigned schedule by using the instruction name (if exist). Then create new call instruction with unassigned schedule and replace it.

Reply:
I was thinking similarly about your last point. There is the potential for a called subroutine not to match the registered schedule with the same name. Doing what you suggest would help with that. If you then replaced a registered schedule, should all the registered schedules calling it be updated as well? We could also have `get_schedule` do the replacement instead of `add_schedule` to help keep things in sync.

I was wondering though if it would be better in the long term to make a new instruction like `CallByName` that stores a name string internally and then `get_schedule` could call `get_schedule` again with the name of the called schedule to replace the `CallByName` with a `Call` to make this desirable behavior of substituting calls more explicit in the schedule definitions.